rfcs/config-luaurc.md

```diff
@@ -0,0 +1,70 @@
+# Configure analysis via .luaurc
+
+## Summary
+
+Introduces a way to configure type checker and linter using JSON-like .luaurc files
+
+## Motivation
+
+While Luau analysis tools try to provide sensible defaults, it's difficult to establish the rules that work for all code.
+For example, some packages may decide that unused variables aren't interesting; other packages may decide that all files should be using strict typechecking mode.
+
+While it's possible to configure some aspects of analysis behavior using --! comments, it can be cumbersome to replicate this in all files.
+
+## Design
+
+To solve this problem, we are going to introduce support for `.luaurc` files for users of command-line Luau tools.
+For a given .lua file, Luau will search for .luaurc files starting from the folder that the .lua file is in; all files in the ancestry chain will be parsed and their configuration
+applied. When multiple files are used, the file closer to the .lua file overrides the settings.
+
+.luaurc is a JSON file that can also contain comments and trailing commas. The file can have the following keys:
+
+- `"mode"`: type checking mode, can be one of "nocheck", "nonstrict", "strict"
+- `"lint"`: lints to enable; points to an array of string literals that correspond to the names of linting rules (see https://luau-lang.org/lint), or `"*"` that means "all rules"
+- `"nolint"`: lints to disable; points to an array of string literals or `"*"`
```

Comment:
Which one of these takes precedence over the other?

Reply:
Within the same file I think it would make sense to error on collisions; otherwise they overwrite the current state inherited from parent.